src/utils/common.ts

```diff
@@ -256,20 +252,24 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
   options: Partial<{
     gasPrice: BigNumberish;
     gasUnits: BigNumberish;
+    baseFeeMultiplier: number;
     transport: Transport;
   }> = {}
 ): Promise<TransactionCostEstimate> {
-  const { gasPrice: _gasPrice, gasUnits, transport } = options || {};
+  const { gasPrice: _gasPrice, gasUnits, baseFeeMultiplier: _baseFeeMultiplier, transport } = options || {};
+  const baseFeeMultiplier = _baseFeeMultiplier ?? 1.0; // A default 1.0x multiplier means no multiplier by default.
 
   const { chainId } = await provider.getNetwork();
   const voidSigner = new VoidSigner(senderAddress, provider);
 
   // Estimate the Gas units required to submit this transaction.
   const queries = [
     gasUnits ? Promise.resolve(BigNumber.from(gasUnits)) : voidSigner.estimateGas(unsignedTx),
-    _gasPrice ? Promise.resolve({ maxFeePerGas: _gasPrice }) : getGasPriceEstimate(provider, chainId, transport),
+    _gasPrice
+      ? Promise.resolve({ maxFeePerGas: _gasPrice })
```

Comment:
Note we don't multiply the gas price if its passed in

Reply:
The Quotes API currently caches the gas price via `getGasCosts -> nativeGasCost`. So with this change is it expected that we should apply the base fee multiplier before we cache it? Or at the time we calculate the fee, i.e. when we call `RelayFeeCalculator.relayerFeeDetails`?

Reply:
You can do either. I think its probably cleaner if you cache the multiplied gas price that way all the API methods can use the same gas price that takes into account the GAS_MARKUP.

But it depends if there are other use cases that read from this cache that do NOT want the base fee multiplied, in that case then you'd only want to cache the base fee without multiplication